node/node.go

```diff
@@ -128,12 +127,17 @@ func newGethNodeConfig(config *params.NodeConfig) (*node.Config, error) {
 			MaxPeers:        config.MaxPeers,
 			MaxPendingPeers: config.MaxPendingPeers,
 		},
-		IPCPath:          makeIPCPath(config),
+		IPCPath:          config.IPCFile,
```

Comment:
I would remove this line completely and put it inside `if config.IPCEnabled {` block so that IPC configuration is in one place.